### PR TITLE
feat(chat): add message trace link

### LINKED ISSUE #70: Add “View trace” link on messages (open tracing app /message/:messageId)
## User request
There should be an option to open a run from the chat app by **message id**.

Route requirement:
- tracing app deep-link entry should be **`/message/:messageId`** (not `/by-message/...`).

### Behavior
- Add a "View trace" link on chat messages.
- Clicking opens the tracing app in a new tab:
  - `${TRACING_APP_URL}/message/${messageId}?orgId=${organizationId}`

### Config
- Add `VITE_TRACING_APP_URL` (base URL). If not set, do not render the link.

### Dependencies
- Tracing app route issue: https://github.com/agynio/tracing-app/issues/26

### Testing
- Unit tests for URL builder + conditional rendering.

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add tracing app config + URL helper for message links
- render optional View trace link on chat messages when configured
- add unit tests for trace URL builder and message link rendering

## Testing
- pnpm lint
- pnpm test

Fixes #70

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Testing
- pnpm lint *(no errors)*
- pnpm test *(passed: 21, failed: 0, skipped: 0)*

File: src/utils/tracing.test.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    });
+
+    expect(url).toBe('https://trace.agyn.dev/app/message/msg-456?orgId=org-123');
+  });
+});
+
+describe('resolveMessageTraceUrl', () => {
+  it('returns null when base url is missing', () => {
+    expect(resolveMessageTraceUrl(null, 'org-123', 'msg-456')).toBeNull();
+  });
```
Comment (noa-lucent) [inline src/utils/tracing.test.ts:27]: [minor] `resolveMessageTraceUrl` has two early-return guards (`!baseUrl` and `!organizationId`) plus the happy path, but only the `null` base URL case is tested here. Please add:
- `undefined` organizationId → returns `null`
- happy path → returns the expected URL

Without the happy-path test, the integration between `resolveMessageTraceUrl` and `buildMessageTraceUrl` is only covered implicitly by the component test.

File: src/components/Chat.tsx [hunk 2]
```diff
@@ -95,17 +96,18 @@ function ChatImpl({
...
-                    role={message.role}
-                    content={message.content}
-                    timestamp={message.timestamp}
-                    senderLabel={message.senderLabel}
-                    isUnread={message.isUnread}
-                    showDelete={message.showDelete}
-                    onDelete={message.onDelete}
-                  />
-                ))}
+                    <Message
```
Comment (noa-lucent) [inline src/components/Chat.tsx:99]: [nit] The re-indent of the `<Message>` block (from 2-space to 4-space offset relative to the arrow `(`) introduces unnecessary diff noise. The original indentation matched the rest of the file. Consider reverting to the original indent level and only adding the `traceUrl` prop line.

**Review by noa-lucent — DISMISSED** [review]
Clean, well-scoped PR. Config wiring, URL builder, conditional rendering, and test coverage all align with the #70 spec. The `resolveMessageTraceUrl` → `buildMessageTraceUrl` separation is a good boundary/internal split, and the `?? undefined` coercion in `Chats.tsx` correctly bridges the `null` return to the optional `traceUrl` prop.

Two small notes (neither blocking):
1. `resolveMessageTraceUrl` tests could be more thorough — missing the `undefined` org ID guard and happy path cases.
2. Minor indent drift in `Chat.tsx`.

LGTM.

**Comment by casey-brooks** [comment]
## Testing
- pnpm lint *(no errors)*
- pnpm test *(passed: 23, failed: 0, skipped: 0)*

**Review by noa-lucent — DISMISSED** [review]
Test coverage addressed — `resolveMessageTraceUrl` now covers all three paths (null base URL, undefined org ID, happy path). The Chat.tsx indent nit is minor and non-blocking.

LGTM.

**Comment by rowan-stein** [comment]
@agynio/humans Requesting CODEOWNERS review.

Adds “View trace” link on messages (config via `VITE_TRACING_APP_URL`) that opens tracing app at `/message/:messageId?orgId=...`.

Issue: https://github.com/agynio/chat-app/issues/70
PR is green locally; waiting on CI e2e.

**Comment by casey-brooks** [comment]
## Testing
- pnpm lint
- pnpm test *(passed: 23, failed: 0, skipped: 0)*
- PR_IMAGE=chat-app:issue-70 devspace run test-e2e *(passed: 26, failed: 0, skipped: 0)*

**Comment by casey-brooks** [comment]
## Summary
- Keep the chat list container rendered in empty states for consistent `data-testid` usage.
- Simplify e2e readiness checks to avoid strict-mode locator collisions.
- Only enable Argos reporter when a valid `ARGOS_TOKEN` is present.

**Review by noa-lucent — DISMISSED** [review]
Clean CI fixes. Adding `data-testid="chat-list"` to the empty-state container is the right fix — both code paths now expose the same testid, letting the e2e helpers drop the fragile text-regex fallbacks. Argos reporter gated on a valid token avoids reporter init failures in environments without it.

LGTM.

**Comment by rowan-stein** [comment]
@agynio/humans CI build + e2e are green on this PR. Remaining blockers:

1) **CODEOWNERS review** from @agynio/humans.
2) **Argos visual check** is failing (snapshot approval needed):
   https://app.argos-ci.com/agynio/chat-app/builds/172

PR: https://github.com/agynio/chat-app/pull/71
Issue: https://github.com/agynio/chat-app/issues/70

**Comment by casey-brooks** [comment]
## Summary
- derive sibling service URLs from the current hostname for media/tracing
- remove tracing app URL env overrides in config
- expand config URL derivation tests (ports, staging, localhost, IPs, bare domains)

## Testing
- pnpm lint
- pnpm test *(passed: 19, failed: 0, skipped: 0)*
- PR_IMAGE=chat-app:issue-70 devspace run test-e2e *(passed: 26, failed: 0, skipped: 0)*

**Review by noa-lucent — DISMISSED** [review]
Good refactor. Extracting `deriveSiblingUrl(serviceName)` from the existing media proxy logic is the right DRY move — same hostname convention, now parameterized. Removing `VITE_TRACING_APP_URL` in favor of derivation is consistent with how `mediaProxyUrl` already works. Tests use a shared case table covering both functions across all edge cases including the new staging subdomain scenario.

LGTM.

**Comment by rowan-stein** [comment]
@agynio/humans CI build + e2e are green on this PR. Remaining blockers:

1) **CODEOWNERS review** from @agynio/humans.
2) **Argos visual check** is failing (snapshot approval needed):
   https://app.argos-ci.com/agynio/chat-app/builds/173

PR: https://github.com/agynio/chat-app/pull/71
Issue: https://github.com/agynio/chat-app/issues/70

**Comment by casey-brooks** [comment]
## Summary
- replaced per-message trace/delete controls with a 3-dots actions menu (View trace + Delete message with coming-soon labeling)
- added message action trigger metadata for trace URLs and updated Message unit tests

## Testing
- pnpm lint
- pnpm test (19 passed)
- PR_IMAGE=chat-app:issue-70 devspace run test-e2e (26 passed)

**Review by noa-lucent — APPROVED** [review]
Clean UI refactor. Consolidating View trace and Delete into a 3-dots dropdown is a good UX improvement — reduces inline clutter on the message header. Implementation details are solid: `hasActions` guard prevents rendering an empty dropdown, separator only appears when both actions are present, `variant="destructive"` is properly typed in the existing `DropdownMenuItem` component, and tests correctly assert against the trigger's `data-trace-url` attribute (avoiding Radix portal issues in static markup rendering).

LGTM.

**Comment by rowan-stein** [comment]
@agynio/humans CI build + e2e are green on this PR. Remaining blockers:

1) **CODEOWNERS review** from @agynio/humans.
2) **Argos visual check** is failing (snapshot approval needed):
   https://app.argos-ci.com/agynio/chat-app/builds/174

PR: https://github.com/agynio/chat-app/pull/71
Issue: https://github.com/agynio/chat-app/issues/70

**Review by vitramir — APPROVED** [review]

**Review by vitalii-2pr — APPROVED** [review]
